src/main/scala/it/unibo/authsim/library/dsl/cryptography/algorithm/symmetric/CaesarCipher.scala

```diff
@@ -0,0 +1,55 @@
+package it.unibo.authsim.library.dsl.cryptography.algorithm.symmetric
+
+import it.unibo.authsim.library.dsl.cryptography.algorithm.SymmetricEncryptionAlgorithm
+/**
+ * Trait for Caesar Cipher algorithm
+ */
+trait CaesarCipher extends SymmetricEncryptionAlgorithm
+
+/**
+ * Companion object of the Caesar Cipher trait
+ */
+object CaesarCipher:
+  /**
+   * Apply method for the object
+   * @return an istance of the Caesar Cipher case class
+   */
+  def apply() = new CaesarCipherImpl()
+  
+  /**
+   * Class representing a basic implementation of the Caesar Cipher algorithm
+   */
+  case class CaesarCipherImpl() extends CaesarCipher:
```

Comment:
If the class is just used to provide a basic implementation of the trait, it should be made private:
```suggestion
  private case class CaesarCipherImpl() extends CaesarCipher:
```